File: 백준/Bronze/13777. Hunt The Rabbit/Hunt The Rabbit.js
const readline = require('readline');

function findRabbit(hiddenNumber) {
    let low = 1;
    let high = 50;
    let attempts = [];

    while (low <= high) {
        let mid = Math.floor((low + high) / 2);
        attempts.push(mid);
        if (mid === hiddenNumber) {
            return attempts;
        } else if (mid < hiddenNumber) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let inputs = [];

rl.on('line', (line) => {
    const number = parseInt(line);
    if (number === 0) {
        rl.close();
    } else {
        inputs.push(number);
    }
}).on('close', () => {
    inputs.forEach(hiddenNumber => {
        const attempts = findRabbit(hiddenNumber);
        console.log(attempts.join(' '));
    });
    process.exit(0);
});

File: 백준/Bronze/13777. Hunt The Rabbit/Hunt The Rabbit.py
import sys

def findRabbit(hiddenNumber):
    low = 1
    high = 50
    attempts = []

    while low <= high:
        mid = (low + high) // 2
        attempts.append(mid)
        if mid == hiddenNumber:
            return attempts
        elif mid < hiddenNumber:
            low = mid + 1
        else:
            high = mid - 1

def main():
    inputs = []
    for line in sys.stdin:
        number = int(line.strip())
        if number == 0:
            break
        inputs.append(number)

    for hiddenNumber in inputs:
        attempts = findRabbit(hiddenNumber)
        print(' '.join(map(str, attempts)))

if __name__ == "__main__":
    main()
